code/modules/networks/computer3/mainframe2/shell.dm

```diff
@@ -221,13 +221,18 @@
 						if (pipetemp)
 							echo_text = pipetemp
 
+						var/add_newline = 1
+						if (command_list[1] == "-n")
```

Comment:
what if it's later

Reply:
If *-n* is later in *command_list*, it echoes as normal.

```
>echo -n abc foo bar
]abc foo bar
```

```
>echo abc foo bar -n
]abc foo bar -n
]
````

```
>echo abc -n foo bar
]abc -n foo bar
]
```